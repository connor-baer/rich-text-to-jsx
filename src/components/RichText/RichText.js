import richTextToJsx from '../../rich-text-to-jsx';

/**
 * Parse and render Contentful rich text to JSX.
 */
const RichText = ({ richText, ...options }) => richTextToJsx(richText, options);

if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  const PropTypes = require('prop-types');

  RichText.propTypes = {
    /**
     * The Contentful rich text AST to be parsed and rendered.
     */
    richText: PropTypes.object,
    /**
     * Map block types to React components, e.g. `p` -> `<Text />`
     * The React component is passed all HTML attributes as props.
     * By default, block types are mapped to the corresponding HTML tags.
     */
    blocks: PropTypes.object,
    /**
     * Override an HTML tag with a React component, e.g. `a` -> `<Link />`
     * The React component is passed all HTML attributes as props.
     */
    inlines: PropTypes.object,
    /**
     * A function create and return a new React element.
     * Defaults to React.createElement()
     */
    createElement: PropTypes.func,
  };
}

/**
 * @component
 */
export default RichText;
