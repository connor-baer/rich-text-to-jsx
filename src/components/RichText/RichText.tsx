import { Document } from '@contentful/rich-text-types';

import richTextToJsx from '../../rich-text-to-jsx';

export interface RichTextProps {
  /**
   * The Contentful rich text AST to be parsed and rendered.
   */
  richText: Document;
  /**
   * Map block types to React components, e.g. `p` -> `<Text />`
   * The React component is passed all HTML attributes as props.
   * By default, block types are mapped to the corresponding HTML tags.
   */
  blocks: object;
  /**
   * Override an HTML tag with a React component, e.g. `a` -> `<Link />`
   * The React component is passed all HTML attributes as props.
   */
  inlines: object;
  /**
   * A function create and return a new React element.
   * Defaults to React.createElement()
   */
  createElement: Function;
}

/**
 * Parse and render Contentful rich text to JSX.
 */
const RichText = ({ richText, ...options }) => richTextToJsx(richText, options);

export default RichText;
