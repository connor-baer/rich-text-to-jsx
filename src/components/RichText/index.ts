import RichText from './RichText';

export default RichText;
