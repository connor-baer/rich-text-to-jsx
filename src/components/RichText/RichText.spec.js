import React from 'react';
import { create } from 'react-test-renderer';
import PropTypes from 'prop-types';

import {
  createDocument,
  headingOne,
  paragraph,
  unorderedList,
  blockquote,
  hr,
} from '../../__fixtures__';

import RichText from './RichText';

describe('RichText', () => {
  describe('styles', () => {
    it('should render with default styles', () => {
      const richText = createDocument([
        headingOne,
        paragraph,
        unorderedList,
        hr,
        blockquote,
      ]);
      const actual = create(<RichText richText={richText} />);
      expect(actual).toMatchSnapshot();
    });

    it('should allow to override normal text', () => {
      const Text = ({ children, ...props }) => (
        <span {...props}>{children}</span>
      );
      Text.propTypes = {
        children: PropTypes.node,
      };

      const richText = createDocument([paragraph]);
      const actual = create(
        <RichText
          richText={richText}
          overrides={{
            text: {
              component: Text,
            },
          }}
        />,
      );
      expect(actual).toMatchSnapshot();
    });
  });
});
