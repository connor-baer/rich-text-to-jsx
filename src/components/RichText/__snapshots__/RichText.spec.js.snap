// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText styles should allow to override normal text 1`] = `
<p>
  <span>
    This is normal text.
  </span>
  <strong>
    This is bold text.
  </strong>
  <em>
    This is italic text.
  </em>
  <u>
    This is underlined text.
  </u>
  <a
    href="https://acme.com"
  >
    <span>
      This is a hyperlink.
    </span>
  </a>
</p>
`;

exports[`RichText styles should render with default styles 1`] = `
Array [
  <h1>
    This is a first-level heading
  </h1>,
  <p>
    This is normal text.
    <strong>
      This is bold text.
    </strong>
    <em>
      This is italic text.
    </em>
    <u>
      This is underlined text.
    </u>
    <a
      href="https://acme.com"
    >
      This is a hyperlink.
    </a>
  </p>,
  <ul>
    <li>
      <p>
        This is normal text.
        <strong>
          This is bold text.
        </strong>
        <em>
          This is italic text.
        </em>
        <u>
          This is underlined text.
        </u>
        <a
          href="https://acme.com"
        >
          This is a hyperlink.
        </a>
      </p>
    </li>
    <li>
      <p>
        This is normal text.
        <strong>
          This is bold text.
        </strong>
        <em>
          This is italic text.
        </em>
        <u>
          This is underlined text.
        </u>
        <a
          href="https://acme.com"
        >
          This is a hyperlink.
        </a>
      </p>
    </li>
    <li>
      <p>
        This is normal text.
        <strong>
          This is bold text.
        </strong>
        <em>
          This is italic text.
        </em>
        <u>
          This is underlined text.
        </u>
        <a
          href="https://acme.com"
        >
          This is a hyperlink.
        </a>
      </p>
    </li>
  </ul>,
  <hr />,
  <blockquote>
    <p>
      This is normal text.
      <strong>
        This is bold text.
      </strong>
      <em>
        This is italic text.
      </em>
      <u>
        This is underlined text.
      </u>
      <a
        href="https://acme.com"
      >
        This is a hyperlink.
      </a>
    </p>
  </blockquote>,
]
`;
